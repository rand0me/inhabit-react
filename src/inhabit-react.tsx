import * as React from "react";

function createInHabitRefHook(client: string) {
    return (el: HTMLDivElement) => {
        if (!el || el.childNodes) {
            return;
        }

        if (document && document.createElement) {
            const script = document.createElement('script');
            script.async = true;
            script.src = "//inhabit-service.cdn.arkadiumhosted.com/Widget/v1/";
            script.dataset.arkClient = client;
            script.dataset.arkLog = "true";
            el.appendChild(script);
        } else {
            el.innerHTML = `<script src="//inhabit-service.cdn.arkadiumhosted.com/Widget/v1/" async data-ark-client=${client} />`;
        }
    }
}

export interface IInhabitProps {
    client: string;
}

export const InHabit = (props: IInhabitProps) =>
    <div ref={createInHabitRefHook(props.client)} />
