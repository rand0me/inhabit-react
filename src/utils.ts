export const createInHabitRefHook = (client: string, url?: string) => (el: HTMLDivElement) => {
    if (!document) {
        el.innerHTML = `<script src="//inhabit-service.cdn.arkadiumhosted.com/Widget/v1/" async data-ark-client=${client} />`
        return
    }
    if (!(el && client)) {
        return
    }
    if (el.querySelector('script')) {
        el.innerHTML = ''
    }
    const script = document.createElement('script')
    script.src = 'https://inhabit-widget-app.azurewebsites.net/js/ark-main.min.js'
    script.dataset.arkClient = client
    if (url) {
        script.dataset.arkContextualUrl = url
    }
    el.appendChild(script)
}