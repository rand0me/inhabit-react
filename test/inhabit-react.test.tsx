const React = require("react");
import { create } from "react-test-renderer";

import { InHabit } from "../src/inhabit-react";

/**
 * Dummy test
 * @jest-environment jsdom
 */
describe("InHabit", () => {
  it("works", () => {
    const vdomTree = create(<InHabit client="test" />).toJSON();
    expect(vdomTree).toMatchSnapshot();
  })
})
