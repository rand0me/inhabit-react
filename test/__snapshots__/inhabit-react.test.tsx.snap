// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InHabit works 1`] = `<div />`;
